manifests/init.pp

```diff
@@ -40,16 +40,16 @@
   Stdlib::Absolutepath $config_file,
   String[1] $package_name,
   String[1] $service_name,
-  String[1] $module_prefix,
-  Stdlib::Absolutepath $init_config_file,
-  Hash $init_config_hash,
+  String[0] $module_prefix,
   String[1] $config_file_header,
-  String[1] $package_ensure,
+  Stdlib::Absolutepath $sbin_path,
+  String[1] $package_ensure = 'installed',
+  Optional[Stdlib::Absolutepath] $init_config_file = undef,
+  Hash $init_config_hash                           = {},
```

Comment:
We also provide a default value which is overridden by existing os-specific info in Hiera and only allows the catalog to be compiled when this unused data is not provided explicitly:

```diff
-  Hash $init_config_hash,
+  Hash $init_config_hash                           = {},
```